File: WindowsMobileClient/Windows Mobile Client/Menus/PasswordContextMenu.vb
Imports System.ComponentModel

<ComponentModel.DesignTimeVisible(True)> _
Public Class PasswordContextMenu : Inherits ContextMenuBase

    Public Sub New()
        MyBase.New(ContextMenuType.Password)
    End Sub

    Protected Overrides Sub OnPopup(ByVal e As System.EventArgs)
        MyBase.OnPopup(e)
        Dim tb As TextBox = TryCast(SourceControl, TextBox)
        If tb IsNot Nothing Then
            PasteMI.Enabled = tb.Enabled AndAlso Not tb.ReadOnly
            ClearMI.Enabled = PasteMI.Enabled
            SelectAllMI.Enabled = PasteMI.Enabled
            Return
        End If
        PasteMI.Enabled = False
        ClearMI.Enabled = False
        SelectAllMI.Enabled = False
    End Sub
End Class

File: Library/activiser.Library.Forms/PasswordContextMenu.cs
using System;
using System.ComponentModel;
using System.Collections.Generic;
using System.Text;
using System.Windows.Forms;

namespace activiser.Library.Forms
{
    [DesignTimeVisible(true), DesignerCategory("Component")]
    public class PasswordContextMenu : ContextMenu
    {

        public PasswordContextMenu()
            : base()
        {
            this.MenuItems.Add(ContextMenuBase.PasteMI);
            this.MenuItems.Add(ContextMenuBase.ClearMI);
            this.MenuItems.Add(ContextMenuBase.SelectAllMI);
        }

        public new MenuItemCollection MenuItems
        {
            get { return base.MenuItems; }
        }

        protected override void OnPopup(System.EventArgs e)
        {
            base.OnPopup(e);
            TextBox tb = SourceControl as TextBox;
            if (tb != null)
            {
                ContextMenuBase.PasteMI.Enabled = tb.Enabled && !tb.ReadOnly;
                ContextMenuBase.ClearMI.Enabled = ContextMenuBase.PasteMI.Enabled;
                ContextMenuBase.SelectAllMI.Enabled = ContextMenuBase.PasteMI.Enabled;
            }
            else
            {
                ContextMenuBase.PasteMI.Enabled = false;
                ContextMenuBase.ClearMI.Enabled = false;
                ContextMenuBase.SelectAllMI.Enabled = false;
            }
        }
    }
}
